frontend/src/components/dashboard/components/categories-list/components/category/category.tsx

```diff
@@ -12,13 +13,33 @@ type Props = {
 };
 
 const Category: FC<Props> = ({ keyName, name }) => {
+  const { searchParams, performSearch } = useSearch();
+  const activeCategory = searchParams.get('category') || '';
+
+  const handleClick = (evt: React.MouseEvent): void => {
+    let category = evt.currentTarget.id;
+
+    if (category === searchParams.get('category')) {
+      category = '';
+    }
+    performSearch('category', category);
+  };
```

Comment:
```suggestion
  const handleClick = (): void => {

    if (keyName === searchParams.get('category')) {
      performSearch('category', '');
      return;
    }
    performSearch('category', keyName);
  };
```

Reply:
agreed, you could just pass keyName of current category to make a closure for onclick handler